### PR TITLE
Integrate Static Code Analysis Tools for IaC files

### PR DESCRIPTION
**Pull request by sbreker**
This pull request enhances build security by integrating the Trivy static code analysis tool configured to scan IaC files.

This new GitHub action is currently configured to scan Dockerfiles only, but this will be expanded in future pull requests.

This PR also addresses a security enhancement identified by Trivy.

### CODE REVIEW

File: .github/workflows/trivy.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  push:
+    branches:
+      - main
+
+permissions: read-all
+
+jobs:
+  trivy:
+    name: Run Trivy for IaC Security Analysis
+    runs-on: ubuntu-20.04
```
Comment (sevein) [inline .github/workflows/trivy.yml]: Can we use 22.04 or 24.04?
* https://github.com/actions/runner-images
* https://ubuntu.com/about/release-cycle#ubuntu

File: .github/workflows/trivy.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
```
Comment (sevein) [inline .github/workflows/trivy.yml]: We can use v4.

**Review by sevein — APPROVED** [review]

File: .github/workflows/trivy.yml (hunk 1 above)
Comment (sbreker) [reply .github/workflows/trivy.yml]: 24.04 in runners is on GitHub's roadmap but not ready yet: https://github.com/github/roadmap/issues/958

Comment (sbreker) [reply .github/workflows/trivy.yml]: Using 22.04.

File: .github/workflows/trivy.yml (hunk 2 above)
Comment (sbreker) [reply .github/workflows/trivy.yml]: Yes - Fixed.

**Pull request merged**